app/controllers/Application.java

```diff
@@ -1939,4 +1896,50 @@ public static Result getTuningParameter(String jobId) {
     logger.info("Size of paramList " + parametersList.size());
     return parametersList;
   }
+
+  private static JsonArray getTuningParameterDetails(List<TuningParameter> parametersList, JobSuggestedParamSet jobSuggestedParamSet, JobSuggestedParamSet userSuggestedParamSet) {
+    JsonArray tuningParameters = new JsonArray();
+    DecimalFormat truncateParamValueFormat = new DecimalFormat("#.##");
+    for (TuningParameter tuningParam : parametersList) {
+      String paramName = tuningParam.paramName;
+      Integer id = tuningParam.id;
+      JobSuggestedParamValue userSuggestedParameterValue = null;
+      if (userSuggestedParamSet != null) {
+        userSuggestedParameterValue = JobSuggestedParamValue.find.select("*")
+            .where()
+            .eq(JobSuggestedParamValue.TABLE.jobSuggestedParamSet + "." + JobSuggestedParamSet.TABLE.id, userSuggestedParamSet.id)
+            .eq(JobSuggestedParamValue.TABLE.tuningParameter + "." + TuningParameter.TABLE.id, id)
+            .findUnique();
+      }
+
+      JobSuggestedParamValue jobSuggestedParameterValue = JobSuggestedParamValue.find.select("*")
+          .where()
+          .eq(JobSuggestedParamValue.TABLE.jobSuggestedParamSet + "." + JobSuggestedParamSet.TABLE.id, jobSuggestedParamSet.id)
+          .eq(JobSuggestedParamValue.TABLE.tuningParameter + "." + TuningParameter.TABLE.id, id)
+          .findUnique();
+
+      if (jobSuggestedParameterValue == null) {
+        logger.info("No Job Suggested Param Value");
+        continue;
+      }
+      JsonObject param = new JsonObject();
+      param.addProperty("paramId",id);
+      param.addProperty("name", paramName);
+      param.addProperty("jobSuggestedParamValue",  truncateParamValueFormat.format(jobSuggestedParameterValue.paramValue));
+      if (userSuggestedParameterValue != null) {
+        param.addProperty("userSuggestedParamValue",  truncateParamValueFormat.format(userSuggestedParameterValue.paramValue));
```

Comment:
Sounds like a code smell. Please see if you can refactor.

Reply:
Please refactor , make sure method doesn't go beyond 50 lines